web-client/src/client/lib.rs

```diff
@@ -678,13 +681,20 @@ impl Client {
             }
         }
 
+        let start_at = if let Some(start_at) = start_at {
+            Some(Blake2bHash::from_str(&start_at)?)
+        } else {
+            None
+        };
+
```

Comment:
Possible, but not sure if better.
```suggestion
        let start_at = start_at.map(|s| Blake2bHash::from_str(&s).ok()).flatten();

```

Reply:
We probably want to fail if `Blake2bHash::from_str` doesn't work. That's why I went for the more complicated approach,